lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1823,10 +1823,18 @@ sub git_log_diff ($self, $dir, $refspec_range, $limit = undef) {
     return $res->{stdout} . $res->{stderr};
 }
 
-sub git_diff ($self, $dir, $refspec_range) {
+sub git_diff ($self, $dir, $refspec_range, $limit = undef) {
     return "Invalid range $refspec_range" if $refspec_range =~ m/UNKNOWN|unreadable git hash/;
     my $timeout = OpenQA::App->singleton->config->{global}->{job_investigate_git_timeout} // 20;
-    my $res = run_cmd_with_log_return_error(['timeout', $timeout, 'git', '-C', $dir, 'diff', '--stat', $refspec_range],
+    my $res = run_cmd_with_log_return_error(['git', '-C', $dir, 'rev-list', '--count', $refspec_range]);
+    if ($res->{return_code}) {
+        return $res->{stdout} . $res->{stderr};
+    }
+    chomp(my $count = $res->{stdout});
+    if ($count > $limit) {
+        return "Too many commits ($count) to create a diff";
+    }
```

Comment:
```suggestion
    return "Too many commits ($count) to create a diff" if $count > $limit;
```

Reply:
Maybe the error message could also state the limit.

Reply:
done